src/ArrayComparator.php

```diff
@@ -124,4 +126,13 @@ private function indent(string $lines): string
     {
         return trim(str_replace("\n", "\n    ", $lines));
     }
+
+    private function canonicalize(array &$array): void
+    {
+        if (array_is_list($array)) {
+            sort($array);
+        } else {
+            ksort($array);
+        }
```

Comment:
Should we traverse into elements here when they are `array`s as well?

Reply:
No, we only want to sort the current array at this point.